### DISCUSSION TITLE
How to know what is holding "Reconciliation in progress"?

### DISCUSSION THREAD

**Discussion by SebTardif** [fluxcd/flux2#1815] [Q&A] [answered]
When I do: `kubectl get -A helmrepositories.source.toolkit.fluxcd.io` one of the hr stay in "Reconciliation in progress", and seem to stall in that state for ever. However, for what I can tell when I'm looking at everything on that application, pod, daemonset, ... all done deploying, and all working. 

1- How can I tell the reason of the hold?
2- How can I force it to be completed?

The big problem in my scenario is that I have other hr that depend on that hr, and so the useless 'in progress' is blocking my most important applications.

**Comment by kingdonb** [comment, marked as answer]
There is an ongoing refactoring effort that will incorporate some Controller Runtime Best-Practices advice the Flux team has learned along the way, to use Status Conditions to report better about intermediary states and help the user understand what's happening between Pending and Ready states.

This will land soon, however in the mean time I would advise you to make sure your `spec.timeout` is set to something less than `spec.interval` – as we have observed this issue that when these values are set to the same number (they are by default) there can be a gap in error reporting, as it just starts reconciling again at the end of the interval and never finishes reporting errors from timeout.
